Resources/Catalog/FileCatalog.py

```diff
@@ -300,50 +314,53 @@ def removeCatalog( self, catalogName ):
 
   def _getSelectedCatalogs( self, desiredCatalogs ):
     for catalogName in desiredCatalogs:
-      res = self._getCatalogConfigDetails( catalogName )
-      if not res['OK']:
-        return res
-      catalogConfig = res['Value']
-      res = self._generateCatalogObject( catalogName )
-      if not res['OK']:
-        return res
-      oCatalog = res['Value']
-      self.readCatalogs.append( ( catalogName, oCatalog, True ) )
-      self.writeCatalogs.append( ( catalogName, oCatalog, True ) )
+      result = self._getCatalogConfigDetails( catalogName )
+      if not result['OK']:
+        return result
+      catalogConfig = result['Value']
+      result = self._generateCatalogObject( catalogName )
+      if not result['OK']:
+        return result
+      oCatalog = result['Value']
+      self.readCatalogs.append( ( catalogName, oCatalog, catalogConfig['Master'] ) )
+      self.writeCatalogs.append( ( catalogName, oCatalog, catalogConfig['Master'] ) )
     return S_OK()
 
-  def _getCatalogs( self ):
+  def _getEligibleCatalogs( self ):
```

Comment:
Connection details are in Resources, those are invariant with respect to VO, setup. All the operational details ( read/write, active/inactive, master ) are in Operations.